amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/fanout/FanOutRecordsPublisher.java

```diff
@@ -184,29 +198,55 @@ private void handleFlowError(Throwable t) {
         }
     }
 
-    private String throwableCategory(Throwable t) {
+    private enum ThrowableType {
+        ACUIRE_TIMEOUT("AcquireTimeout"), READ_TIMEOUT("ReadTimeout"), UNKNOWN("Unknown");
```

Comment:
Probably should be `OTHER` vs `UNKNOWN`.  Unknown feels like it's a failure state (subjective I know).

Reply:
Thought naming it other earlier, but ended up naming it unknown. Changed it to other.